src/torrent.cpp

```diff
@@ -3394,20 +3397,25 @@ namespace libtorrent
 			{
 				// this is an i2p name, we need to use the sam connection
 				// to do the name lookup
-				/*
-				m_ses.m_i2p_conn.async_name_lookup(i->ip.c_str()
-					, boost::bind(&torrent::on_i2p_resolve
-					, shared_from_this(), _1));
-				*/
-				// it seems like you're not supposed to do a name lookup
-				// on the peers returned from the tracker, but just strip
-				// the .i2p and use it as a destination
-				std::string hostname = i->hostname.substr(i->hostname.size() - 4);
-				torrent_state st = get_policy_state();
-				need_policy();
-				if (m_peer_list->add_i2p_peer(hostname.c_str(), peer_info::tracker, 0, &st))
-					state_updated();
-				peers_erased(st.erased);
+				if (boost::algorithm::ends_with (i->hostname, ".b32.i2p"))
+					r.i2pconn->async_name_lookup (i->hostname.c_str ()
```

Comment:
now that there's a new async_*() call, there needs to be a corresponding "add_outstanding_async" call too. The name that's passed in must match the name at the top of the handler function (in this case torrent::on_i2p_resolve()).